### PR TITLE
Fix error in EditResource 

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

This bug doesn't exist on master because of #5195

### CODE REVIEW

File: src/renderer/components/dock/edit-resource/view.tsx [hunk 1]
```diff
@@ -45,8 +45,8 @@ class NonInjectedEditResource extends React.Component<EditResourceProps & Depend
         const tabData = this.props.editResourceStore.getData(this.props.tab.id);
         const obj = this.resource;
 
-        if (!obj) {
-          if (store?.isLoaded) {
+        if (!obj && store) {
+          if (store.isLoaded) {
             // auto-close tab when resource removed from store
             this.props.closeTab(this.props.tab.id);
           } else if (!store.isLoading) {
```
Comment (Nokel81) [inline src/renderer/components/dock/edit-resource/view.tsx:52]: The actual error reported was on this line.

**Review by ixrock — APPROVED** [review]

File: src/renderer/components/dock/edit-resource/view.tsx [hunk 2]
```diff
@@ -45,8 +45,8 @@ class NonInjectedEditResource extends React.Component<EditResourceProps & Depend
         const tabData = this.props.editResourceStore.getData(this.props.tab.id);
         const obj = this.resource;
 
-        if (!obj) {
-          if (store?.isLoaded) {
+        if (!obj && store) {
+          if (store.isLoaded) {
```
Comment (jim-docker) [inline src/renderer/components/dock/edit-resource/view.tsx:49]: So the bug was that if store was null/undefined then execution shouldn't have gone to the `else` block? These are tricky to see, and made more common by the convenience of the `?.` operator

**Review by jim-docker — APPROVED** [review]

File: src/renderer/components/dock/edit-resource/view.tsx (hunk 2 above)
Comment (Nokel81) [reply src/renderer/components/dock/edit-resource/view.tsx:49]: No the bug was that if the `store` was `null` or `undefined` nothing should have happened, but instead it did go to the `else` block.

Comment (jim-docker) [reply src/renderer/components/dock/edit-resource/view.tsx:49]: thanks (I meant shouldn't, corrected)

**Pull request merged**